wagtail/snippets/models.py

```diff
@@ -83,51 +83,33 @@ def register_snippet(registerable, viewset=None):
 def _register_snippet_immediately(registerable, viewset=None):
```

Comment:
This could be further simplified if we stopped supporting the following registration style:

```py
register_snippet(MyModel, viewset=MyModelViewSet)
```

And instead just support the following styles:

```py
# Old style:
register_snippet(MyModel)
# equivalent to
@register_snippet
class MyModel(models.Model):
    ....

# With custom ViewSet
register_snippet(MyModelViewSet)

# Optionally, we can also support instances, not just classes
# (this is already done in this PR)
register_snippet(SomeGenericViewSet(model=MyModel, ...))
```

The `viewset=` parameter was only documented for some time when custom `SnippetViewSet` support was added but before the custom menu registration was added, so 4.1.x and 4.2.x releases. As of 5.0.x, the docs no longer mentions `viewset=` in favour of passing the viewset to `register_snippet` as the first parameter and setting the model on the viewset.